### PR TITLE
idris2: init at 0.0.0 (bf5b229)

### PR DESCRIPTION
**Pull request by lodi**
###### Motivation for this change

Added Idris2 compiler.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/idris2/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  pname = "idris2";
+
+  src = fetchFromGitHub {
+    owner = "edwinb";
+    repo = "Idris2";
+    rev = "bf5b2298e068ebeaddd70b695426f52ebe67c1ab";
+    sha256 = "1ixmdy4zbllmpwmvmmqv1rj9a2y243pggf42pdqkvrkjw50zqlyj";
+  };
+
+  version = "0.0.0";
```
Comment (tobim) [inline pkgs/development/compilers/idris2/default.nix]: Nixpkgs typically uses the YYYY-MM-DD schema for untagged sources.

**Review by tobim — CHANGES_REQUESTED** [review]
Builds and installs fine locally.

It might be a good idea to add this to idrisPackages set.

Related: #71125

File: pkgs/development/compilers/idris2/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# racket is busted: https://github.com/NixOS/nixpkgs/issues/11698
+
+{ stdenv, fetchFromGitHub, makeWrapper, runCommand
+, idris, chez, racket, chickenPkgs
+}:
+
+stdenv.mkDerivation rec {
```
Comment (tobim) [inline pkgs/development/compilers/idris2/default.nix]: Is there a specific reason you did not use the `build-idris-package` function here?

File: pkgs/development/compilers/idris2/default.nix (hunk 1 above)
Comment (lodi) [reply pkgs/development/compilers/idris2/default.nix]: Ok, updated to date-based version format.

File: pkgs/development/compilers/idris2/default.nix (hunk 2 above)
Comment (lodi) [reply pkgs/development/compilers/idris2/default.nix]: I wasn't sure how to make this work with build-idris-package.  E.g. idris2 tests are in their own ipkg called from the makefile; buildInputs might be set incorrectly (idris should only be a nativeBuildInput for idris2 I think?); etc.

**Comment by locallycompact** [comment]
Hi is this still active? Idris 2 has just released a v0.1.0.

https://www.idris-lang.org/idris-2-version-010-released.html

I can pick this up.

**Comment by lodi** [comment]
Sorry for the delay @locallycompact.  I updated the PR to track the latest Idris2 git rev.

I think this PR was never merged because it doesn't integrate with the existing idris-packages infrastructure.  I'm not too familiar with how idris-packages works so this PR kind of stalled out.  If you can get this accepted to nixpkgs in one way or another that'd be great!

**Comment by locallycompact** [comment]
@lodi Hi, thanks for updating this.

I guess then, anybody watching, what are the criteria for integrating it with idris-packages?

How does say python2 and python3 handle this?

**Comment by Mic92** [comment]
Is there anything in this PR that we don't have in https://github.com/NixOS/nixpkgs/pull/88645 ?

**Comment by lodi** [comment]
Nothing major no; just some minor difference to support Chicken Scheme and run tests at build time.  I'll close this PR and contribute to #88645 instead.

**Pull request closed without merging**